dspace-server-webapp/src/main/java/org/dspace/app/rest/utils/Utils.java

```diff
@@ -600,20 +601,29 @@ private void embedMethodLevelRel(HALResource<? extends RestAddressableModel> res
 
     /**
      * Wraps the given linked object (retrieved from a link repository or link method on the rest item)
-     * in an object that is appropriate for embedding, if needed.
+     * in an object that is appropriate for embedding, if needed. Does not perform the actual embed; the
+     * caller is responsible for that.
      *
+     * @param resource the resource from which the embed will be made.
      * @param linkedObject the linked object.
      * @param link the link, which is used if the linked object is a list or page, to determine the self link
      *             and embed property name to use for the subresource.
-     * @return the wrapped object.
+     * @return the wrapped object, which will have an "embed level" one greater than the given parent resource.
      */
-    private Object wrapForEmbedding(Object linkedObject, Link link) {
+    private Object wrapForEmbedding(HALResource<? extends RestAddressableModel> resource,
+                                    Object linkedObject, Link link) {
+        int childEmbedLevel = resource.getContent().getEmbedLevel() + 1;
         if (linkedObject instanceof RestAddressableModel) {
-            return converter.toResource((RestAddressableModel) linkedObject);
+            RestAddressableModel restObject = (RestAddressableModel) linkedObject;
+            restObject.setEmbedLevel(childEmbedLevel);
+            return converter.toResource(restObject);
```

Comment:
Jira ticket for the latter, so we don't lose track of it?

Reply:
Interesting. I also can't say why a query would have been run for the bundles and communities of the logo bitstreams: The code that limits the level (in Utils.embedRelFromRepository) definitely returns *before* the link repository is called. And Bitstream.java specifies FetchType.LAZY for both relations.

This seems worth following up on at some point to identify the exact cause, as it may be systemic.